sov-modules/sov-modules-api/src/mocks.rs

```diff
@@ -46,8 +49,10 @@ impl TryFrom<String> for MockPublicKey {
 
 impl PublicKey for MockPublicKey {
     fn to_address<A: AddressTrait>(&self) -> A {
-        let pub_key_hash = <MockContext as Spec>::Hasher::hash(&self.pub_key);
-        A::try_from(&pub_key_hash).expect("todo")
+        let mut hasher = sha2::Sha256::new();
```

Comment:
Replaced with concrete hasher, so it does not depend on MockContext as Spec, because it brings db dependency from storage type

Reply:
This isn't ideal - we want to be able to configure the hash function (for efficiency in a zkvm), and it will be surprising to future developers when changing the `Hash` in `Context` has no effect. Can we feature-gate the db dependency instead?

Reply:
Done!